flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcSplitReaderUtil.java

```diff
@@ -158,8 +158,8 @@ static TypeDescription logicalTypeToOrcType(LogicalType type) {
 			case DECIMAL:
 				DecimalType decimalType = (DecimalType) type;
 				return TypeDescription.createDecimal()
-						.withPrecision(decimalType.getPrecision())
-						.withScale(decimalType.getScale());
+						.withScale(decimalType.getScale())
```

Comment:
Just curious why this order matters?

Reply:
I had some explanations in the description of this PR. In short, it's because ORC `TypeDescription` assumes default scale to be 10, and asserts `scale < precision` when we set the precision.
It'd be better if they provide an interface to set precision and scale in one go.

Reply:
```
public TypeDescription withPrecision(int precision) {
    if (precision < 1 || precision > MAX_PRECISION || **scale > precision**){
      throw new IllegalArgumentException("precision " + precision +
          " is out of range 1 .. " + scale);
    }
    this.precision = precision;
    return this;
  }
```
There is check in `withPrecision`, and default scale is 10.